api/request.go

```diff
@@ -157,6 +157,27 @@ type SliceRequest struct {
 
 	// Line number of the slice
 	Lineno *int `json:"lineno" binding:"required" example:"10000"`
+
+	// Restrict the slice in the other dimensions (sub-slicing)
+	//
+	// Bounds can be used to retrieve sub-slices. For example: when requesting
+	// a time slice you can set inline and/or crossline bounds to restrict the
+	// area of the slice.
+	//
+	// Bounds are optional. Not specifying any bounds will produce the requested
+	// slice through the entire volume.
+	//
+	// Any bounds in the same direction as the slice itself are ignored.
+	//
+	// Bounds are applied one after the other. If there are multiple bounds in the
+	// same direction, the last one takes precedence.
+	//
+	// Bounds will throw out-of-bounds error if their range is outside the
```

Comment:
Yes, will add test.

I think errors are the right choice here. The information needed to set correct bounds are all available from the metadata endpoint. 

On the required min/max. I did that to keep it simple for know and I didn't really know if's a need. If you think it is, I can add. Not by this `-1` thing tho. Instead make the Upper and Lower fields `*float32` and remove required. Then I think it would be nicer to let go check if they are not set and if so set to the extent of the file. (Rather than propagating optionals through C, which is ugly).

Reply:
If you think it is easy to make them optional, do it. If not, then we wait until first person comes and complains.

I think Renaud wasn't himself sure this would be useful.

Reply:
I'll wait and make an issue for it